eng/workloads/workloads.csproj

```diff
@@ -0,0 +1,168 @@
+      <Output TaskParameter="SwixProjects" ItemName="SwixProjects" />
+      <Output TaskParameter="Msis" ItemName="Msis" />
+    </GenerateVisualStudioWorkload>
+
+    <!-- Build all the SWIX projects. This requires full framework MSBuild-->
+    <MSBuild Projects="@(SwixProjects)" Properties="SwixBuildTargets=$(SwixBuildTargets);ManifestOutputPath=$(VersionedVisualStudioSetupInsertionPath)" />
+
+    <!-- Gather .wixobj files for post-build signing. We'll have to batch since we generated multiple MSIs in the previous step. -->
+    <MSBuild Projects="$(MSBuildProjectFile)" Properties="_WixObjDir=%(Msis.WixObj);_Msi=%(Msis.Identity)" Targets="CreateWixPack" />
+
+    <!-- Build the Visual Studio manifest project. -->
+    <ItemGroup>
+      <VisualStudioManifestProjects Include="OptionalWorkloads.vsmanproj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(VisualStudioManifestProjects)" Properties="SwixBuildTargets=$(SwixBuildTargets);ManifestOutputPath=$(VersionedVisualStudioSetupInsertionPath);OutputPath=$(VersionedVisualStudioSetupInsertionPath)" />
+
+    <!-- Build all the MSI payload packages for NuGet. -->
+    <ItemGroup>
+      <MsiPackageProjects Include="%(Msis.PackageProject)" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(MsiPackageProjects)" Properties="OutputPath=$(ArtifactsShippingPackagesDir)" Targets="restore;pack" />
+  </Target>
+
+  <!-- Target to create a single wixpack for signing -->
+  <Target Name="CreateWixPack">
+    <ItemGroup>
+      <_WixObj Include="$(_WixObjDir)\**\*.wixobj" />
+    </ItemGroup>
+
+    <CreateLightCommandPackageDrop
+      LightCommandWorkingDir="$(LightCommandObjDir)"
+      OutputFolder="$(LightCommandPackagesDir)"
+      NoLogo="true"
+      Cultures="en-us"
+      InstallerFile="$(_Msi)"
+      WixExtensions="WixUIExtension;WixDependencyExtension;WixUtilExtension"
+      WixSrcFiles="@(_WixObj)">
+      <Output TaskParameter="OutputFile" PropertyName="_LightCommandPackageNameOutput" />
+    </CreateLightCommandPackageDrop>
+  </Target>
+
+  <!-- These are just individual targets for testing local builds. -->
+  <Target Name="BuildSwixProjects">
+    <ItemGroup>
+      <SwixProjects Include="$(WorkloadIntermediateOutputPath)**\*.swixproj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(SwixProjects)" BuildInParallel="true" Properties="SwixBuildTargets=$(SwixBuildTargets);ManifestOutputPath=$(VersionedVisualStudioSetupInsertionPath)" />
+  </Target>
+
+  <Target Name="BuildVisualStudioManifest">
+    <ItemGroup>
+      <VisualStudioManifestProjects Include="Microsoft.NET.vsmanproj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(VisualStudioManifestProjects)" BuildInParallel="true" Properties="SwixBuildTargets=$(SwixBuildTargets);ManifestOutputPath=$(VersionedVisualStudioSetupInsertionPath);OutputPath=$(VersionedVisualStudioSetupInsertionPath)" />
+  </Target>
+
+  <Target Name="GenerateVersions" DependsOnTargets="GetAssemblyVersion">
```

Comment:
This block of code came from my original prototype insertion into dotnet/installer. The big issue it tried to address was being able to assign incremental versioning to the components that represent a specific workload so the sequential insertions into VS would have different component versions. Normally, we'd author components in the VS repo, so they would match the version of VS itself. 

If this does work in emsdk, then we can keep it, but if emsdk had a versioning scheme (non-semver) that was maybe used to version files (major.,minor.patch.build that .NET used), we could use that. 

Just wanted to provide some context on where this came from.